falco.yaml

```diff
@@ -764,6 +773,7 @@ metrics:
   output_rule: true
   # output_file: /tmp/falco_stats.jsonl
   resource_utilization_enabled: true
```

Comment:
what about having a list of metrics instead of many boolean values? maybe something like 
```yaml
enabled_metrics: ["kernel_countes", ... ]
```
if we like the idea we could deprecate the old flags and introduce this new config. If we go in this way probably I would merge `libbpf_stats_enabled` into `kernel_event_counters_enabled`, in the end, to have the `libbpf_stats_enabled` you need to enable the stats `sudo sysctl -w kernel.bpf_stats_enabled=1`, so we already have a knob. Moreover, it is something only related to `bpf` so maybe it is better to be more agnostic and hide it inside `kernel_countes` WDYT?

Reply:
We can discuss a `enabled_metrics` list. Given we are deprecating more urgent old settings I am not be sure if this is worth it in the near-term. Since metrics is now "Stable" we would need to follow a deprecation cycle. I would propose to open a discussion issue, tag all libs and falco maintainers and if the majority votes in favor of `enabled_metrics: ["kernel_countes", ... ]` start the proper deprecation cycle. WDYT?

Kernel settings can be shared amongst teams deploying their tools. Therefore, it can be fair to assume that maybe another team enabled libbpf stats, but you still have a use case of not wanting to collect libbpf stats for Falco. I would prefer giving the end user full control and be very selective with the metrics especially given it's quite a lot of metrics and perhaps you only need some of them sometimes. With that I would also not merge libbf stats and the kernel event counters. All these reasons were the motivation to have these flags that are all teh way pushed down to libs in the first place.

Reply:
yeah we can discuss it without blocking the PR, i will tag all here so even if we merge this we can discuss with the clear scope of this PR @falcosecurity/falco-maintainers 

Re `libbpf_stats_enabled`: my idea was to use the sysctl option `kernel.bpf_stats_enabled` as a knob, so if a team wants to enable libbpf stats together with the kernel stats it has just to type a `sudo sysctl -w kernel.bpf_stats_enabled=1` on the node, while if they want to disable them they could use `sudo sysctl -w kernel.bpf_stats_enabled=0`. The libsinsp code at every interval could check `kernel.bpf_stats_enabled` (or maybe just at init time, it depends on what we want) and add the libbpf metrics if needed. BTW it was just an idea, no strong opinion here

Reply:
I agree about using `sysctl` value `kernel.bpf_stats_enabled` to chose whether to enable/disable our kernel stats without further configuration by the user, it's a great idea actually! We could use its value as a default value for the configuration key and ship it commented? (same for `kernel_event_counters_enabled` eventually)

> We can discuss a enabled_metrics list

I don't have a strong opinion on this one; i think from an UX perspective it is better to already have all the keys and just set "true"/"false" whether i need them, than to add strings inside a set.